dask/dataframe/groupby.py

```diff
@@ -280,6 +281,40 @@ def _groupby_get_group(df, by_key, get_key, columns):
         return df.iloc[0:0]
 
 
+def numeric_only_deprecate_default(func):
+    """Decorator for methods that should warn when numeric_only is default"""
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        numeric_only = kwargs.get("numeric_only", no_default)
+        if PANDAS_GT_150 and not PANDAS_GT_200 and numeric_only is no_default:
+            warnings.warn(
+                "The default value of numeric_only will be `False` "
+                "in a future version of Pandas.",
+                FutureWarning,
+            )
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+def numeric_only_not_implemented(func):
+    """Decorator for methods that can't handle numeric_only=False"""
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        numeric_only = kwargs.get("numeric_only", no_default)
+        if numeric_only is False or (PANDAS_GT_200 and numeric_only is no_default):
+            raise NotImplementedError(
+                "'numeric_only=False' is not implemented in Dask."
+            )
+        if numeric_only is True:
+            self.obj = self.obj._get_numeric_data()
```

Comment:
It seems strange to mutate `self.obj` here. This might even lead to issues if other groupby operations are called subsequently. 

Why do we need to select numeric data here in the first place? Are there cases where passing `numeric_only=True` as a chunk/aggregation kwarg isn't sufficient?

Reply:
Good catch. I was experimenting with filtering out non-numeric data in the decorator, but I wasn't going to keep this piece.